src/main/java/org/candlepin/subscriptions/inventory/db/model/InventoryHost.java

```diff
@@ -150,7 +150,7 @@ from jsonb_array_elements(h.system_profile_facts->'installed_products') as items
            and (h.system_profile_facts->>'host_type' IS NULL OR h.system_profile_facts->>'host_type' <> 'edge')
            and NOW() < stale_timestamp + make_interval(days => :culledOffsetDays)
         -- NOTE: ordering is crucial for correct streaming reconciliation of HBI data
-        order by hardware_subman_id, any_hypervisor_uuid, inventory_id
+        order by hardware_subman_id, any_hypervisor_uuid, inventory_id, provider_id
```

Comment:
I think this can work, but we should put provider_id first for it to have an impact. (Because this query should not return any duplicate inventory_id values).

Reply:
I don't follow why we should put the provider_id first to have an impact. Do you mean before `hardware_subman_id`? or `inventory_id`?